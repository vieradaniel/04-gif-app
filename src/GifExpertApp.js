import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp =({ defaultCategories = []})=>{


    //const [categories, setCategories] = useState(['Sailor Moon']);
    const [categories, setCategories] = useState( defaultCategories );

    // const addCatHandler= ()=>{

    //     //setCategories ([...categories,'Sailor Moon']);
    //     setCategories (cats => [...categories,'Sailor Moon']);

    // }

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories ={ setCategories }/>
            <hr />
           
            <ol>
                { 
                    categories.map(category =>(
                        <GifGrid 
                            key={ category }
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
        
    )
}

export default GifExpertApp;