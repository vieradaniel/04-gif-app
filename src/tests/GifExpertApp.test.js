import React from 'react';
import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';




describe('testing <GifExpertApp/>', () => {
  
    test('should show the correct component and match with the snapshot', () => {
      
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('should show a list of categories', () => {
      const categories = ['sailor moon','dragon ball'];
      const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    
    

});
