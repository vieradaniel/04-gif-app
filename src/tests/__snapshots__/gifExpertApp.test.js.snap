// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GifExpertApp/> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="sailor moon"
      key="sailor moon"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`testing <GifExpertApp/> should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
